computer_vision/cnn_classification/cnn_classification.py

```diff
@@ -0,0 +1,91 @@
+"""
+Convolutional Neural Network
+
+Objective : To train a CNN model detect if TB is present in Lung X-ray or not.
+
+Resources CNN Theory : https://en.wikipedia.org/wiki/Convolutional_neural_network
+Resources Tensorflow : https://www.tensorflow.org/tutorials/images/cnn
+
+Download dataset from :
+https://lhncbc.nlm.nih.gov/LHC-publications/pubs/TuberculosisChestXrayImageDataSets.html
+
+"""
+
+# Part 1 - Building the CNN
+
+# Importing the Keras libraries and packages
+import tensorflow as tf
+from tensorflow.keras import layers, models
+
+import numpy as np
+
+
+# Initialising the CNN
+classifier = models.Sequential()
+
+# Step 1 - Convolution
+classifier.add(layers.Conv2D(32, (3, 3), input_shape=(64, 64, 3),
+                             activation="relu"))
+
+# Step 2 - Pooling
+classifier.add(layers.MaxPooling2D(pool_size=(2, 2)))
+
+# Adding a second convolutional layer
+classifier.add(layers.Conv2D(32, (3, 3), activation="relu"))
+classifier.add(layers.MaxPooling2D(pool_size=(2, 2)))
+
+# Step 3 - Flattening
+classifier.add(layers.Flatten())
+
+# Step 4 - Full connection
+classifier.add(layers.Dense(units=128, activation="relu"))
+classifier.add(layers.Dense(units=1, activation="sigmoid"))
+
+# Compiling the CNN
+classifier.compile(optimizer="adam", loss="binary_crossentropy",
+                   metrics=["accuracy"])
+
+# Part 2 - Fitting the CNN to the images
+
+
+# Load Trained model weights
+
+# from keras.models import load_model
+# regressor=load_model('cnn.h5')
+
+train_datagen = tf.keras.preprocessing.image.ImageDataGenerator(
+    rescale=1.0 / 255, shear_range=0.2, zoom_range=0.2, horizontal_flip=True
+)
+
+test_datagen = tf.keras.preprocessing.image.ImageDataGenerator(
+    rescale=1.0 / 255)
+
+training_set = train_datagen.flow_from_directory(
+    "dataset/training_set", target_size=(64, 64),
+    batch_size=32, class_mode="binary"
+)
+
+test_set = test_datagen.flow_from_directory(
+    "dataset/test_set", target_size=(64, 64),
+    batch_size=32, class_mode="binary"
+)
+
+classifier.fit_generator(
+    training_set, steps_per_epoch=5, epochs=30, validation_data=test_set
+)
+
+classifier.save("cnn.h5")
+
+# Part 3 - Making new predictions
+
+
+test_image = tf.keras.preprocessing.image.load_img(
+    "dataset/single_prediction/image.png", target_size=(64, 64))
```

Comment:
Is this the dataset which you referred in the module docstring?

> Download dataset from :
https://lhncbc.nlm.nih.gov/LHC-publications/pubs/TuberculosisChestXrayImageDataSets.html

If so, can you provide some instruction steps like how to download and where to put it so that this script can pick it up, make it as simple as possible for new users :)

Reply:
I have added steps on working the dataset in the docstring, rest information is available on keras and tensorflow website which is already mentioned